### PR TITLE
[13.0.x] ISPN-13616 Stream pipeline blocks if filterKeys() gets an empty set

### PR DESCRIPTION
**Pull request by wburns**
https://issues.redhat.com/browse/ISPN-13616

### CODE REVIEW

**Comment by wburns** [comment]
Replaces #9786

**Comment by pruivo** [comment]
@wburns don't forget to update this PR :)

**Comment by wburns** [comment]
> @wburns don't forget to update this PR :)

Yup, was just waiting on main. I will update this momentarily.

**Comment by wburns** [comment]
Updated, sorry took longer as I was looking at the scattered test failure until I realized it fails all the time on 13.0.x.

File: core/src/main/java/org/infinispan/reactive/publisher/impl/PublisherHandler.java [hunk 1]
```diff
@@ -173,6 +173,32 @@ private void closePublisher(String requestId, PublisherState state) {
       }
    }
 
+   public static class SegmentResult {
```
Comment (pruivo) [inline core/src/main/java/org/infinispan/reactive/publisher/impl/PublisherHandler.java:176]: unsued and can be removed.

Comment (wburns) [reply core/src/main/java/org/infinispan/reactive/publisher/impl/PublisherHandler.java:176]: Oh whoops, leftovers from main.

**Comment by wburns** [comment]
Updated an pushed a commit I missed :facepalm:

**Pull request merged**